packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -276,6 +276,14 @@ export interface FunctionOptions extends EventInvokeConfigOptions {
    */
   readonly events?: IEventSource[];
 
+  /**
+   * Whether the associated CloudWatch log group should be automatically deleted
+   * when the function is removed from the stack or when the stack is deleted.
+   *
+   * @default false
+   */
+  readonly autoDeleteLogGroup?: boolean;
```

Comment:
This is inconsistent with our contract elsewhere when the user sets the Removal Policy. Please see other places where we have done this and follow that model. You use the enum below.

Reply:
Hi @TheRealAmazonKendra thank you for the review!

This API is actually inspired by [`s3.Bucket.autoDeleteObjects`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.Bucket.html#autodeleteobjects). In this case the only concern for developers is whether the log group is automatically deleted or not. It is only a binary choice and there is no need to use RemovalPolicy enum to specify that. One of the main advantages of CDK constructs is abstraction, and here we abstract away RemovalPolicy to provide better developer experience. 

Also internally we do not use CloudFormation RemovalPolicy here but using a feature implemented in the LogRetention custom resource below. I'm not sure if it is inconsistent to use the different contract from other CFn removal policies. Isn't it natural that different things will have different contracts?

https://github.com/aws/aws-cdk/blob/3e55092fb70e0ec74ee7c4144d6e39a29d8757ae/packages/%40aws-cdk/aws-logs/lib/log-retention.ts#L102

I'm open to discussion about this :) currently I don't see much benefits to use enum here though.